pex/compatibility.py

```diff
@@ -85,6 +92,7 @@ def exec_function(ast, globals_map):
 
 
 else:
+    # This will result in `exec_function` being defined at runtime.
```

Comment:
I'm concerned that this will mean type check errors wherever it's used. Is that the case? If so I feel like a no-op might make sense. But if it's used in only one or two places then this is fine.

Reply:
I think not redefining the function with a no-op (as you've done here) is probably safer, in case it messes up how the function is defined. Any type ignores can be removed after dropping py2.

Reply:
> I think not redefining the function with a no-op (as you've done here) is probably safer, in case it messes up how the function is defined. Any type ignores can be removed after dropping py2.

Exactly. This code was written in 2013 and is only used in one single place. I think it's unlikely we'll start using it more.